tests/impl.cpp

```diff
@@ -9149,14 +9149,71 @@ result_t test_mm_cmpestrc(const SSE2NEONTestImpl &impl, uint32_t iter)
     return TEST_UNIMPL;
 }
 
+typedef struct {
+    int16_t a[8], b[8];
+    int la, lb;
+    const int imm8;
+    int expect;
+} test_mm_cmpestri_data_t;
+static test_mm_cmpestri_data_t test_mm_cmpestri_data = {
+    {13, 6, 5, 4, 3, 2, 1, 3},
+    {-7, 16, 5, 4, -1, 6, 1, 3},
+    10,
+    10,
+    _SIDD_SWORD_OPS | _SIDD_CMP_RANGES | _SIDD_MOST_SIGNIFICANT,
+    7,
+};
+
 result_t test_mm_cmpestri(const SSE2NEONTestImpl &impl, uint32_t iter)
 {
-    return TEST_UNIMPL;
-}
+    int16_t *a = test_mm_cmpestri_data.a, *b = test_mm_cmpestri_data.b;
+    int la = test_mm_cmpestri_data.la, lb = test_mm_cmpestri_data.lb;
+    const int imm8 = test_mm_cmpestri_data.imm8;
+    int expect = test_mm_cmpestri_data.expect;
+    __m128i ma, mb;
+    memcpy(&ma, a, sizeof(ma));
+    memcpy(&mb, b, sizeof(mb));
+#if defined(__arm__) || defined(__aarch64__)
+    int res = _mm_cmpestri(ma, la, mb, lb, imm8);
+#else
+    int res = _mm_cmpestri(
+        ma, la, mb, lb,
+        _SIDD_SWORD_OPS | _SIDD_CMP_RANGES | _SIDD_MOST_SIGNIFICANT);
+#endif
+    return (res == expect) ? TEST_SUCCESS : TEST_FAIL;
+}
+
+typedef struct {
+    int16_t a[8], b[8];
+    int la, lb;
+    const int imm8;
+    int16_t expect[8];
+} test_mm_cmpestrm_data_t;
+static test_mm_cmpestrm_data_t test_mm_cmpestrm_data = {
+    {13, 6, 5, 4, 3, 2, 1, 3},
+    {-7, 16, 5, 4, -1, 6, 1, 3},
+    10,
+    10,
+    _SIDD_SWORD_OPS | _SIDD_CMP_RANGES | _SIDD_MOST_SIGNIFICANT,
+    {0, 0, 0, 0, 0, 0, -1, -1}};
 
 result_t test_mm_cmpestrm(const SSE2NEONTestImpl &impl, uint32_t iter)
 {
-    return TEST_UNIMPL;
+    int16_t *a = test_mm_cmpestrm_data.a, *b = test_mm_cmpestrm_data.b;
+    int la = test_mm_cmpestrm_data.la, lb = test_mm_cmpestrm_data.lb;
+    const int imm8 = test_mm_cmpestrm_data.imm8;
+    int16_t *expect = test_mm_cmpestrm_data.expect;
+    __m128i ma, mb, res;
+    memcpy(&ma, a, sizeof(ma));
+    memcpy(&mb, b, sizeof(mb));
+#if defined(__arm__) || defined(__aarch64__)
```

Comment:
Can you avoid the special cases (Arm/Aarch64) for invoking `_mm_cmpestrm`?

Reply:
If I simply change the statement from:
```
#if defined(__arm__) || defined(__aarch64__)
    res = _mm_cmpestrm(ma, la, mb, lb, imm8);
#else
    res = _mm_cmpestrm(
        ma, la, mb, lb,
        _SIDD_SWORD_OPS | _SIDD_CMP_RANGES | _SIDD_MOST_SIGNIFICANT);
#endif
```
to:
```
res = _mm_cmpestrm(ma, la, mb, lb, imm8);
```
Just like NEON's `vgetq_lane_xxx` case, I will receive a compilation error when testing on x86 platform:
```
error: the fifth argument must be an 8-bit immediate
 9216 |     res = _mm_cmpestrm(ma, la, mb, lb, imm8);
      |           ^~~~~~~~~~~~
```

I think some kinds of macro usages such as X Macro will help me. Let me come up with the solution.